src/components/step/ConnectDeviceAccountStepContainer/ConnectDeviceAccountStepContainer.tsx

```diff
@@ -2,20 +2,75 @@ import React, { useState, useEffect } from "react";
 import MyDataHelps, { StepConfiguration } from "@careevolution/mydatahelps-js";
 import ConnectDeviceAccountStep from "../ConnectDeviceAccountStep";
 
-export default function () {
+export interface ConnectDeviceAccountStepContainerProps {
+    deviceType?: string;
+}
+
+export default function (props: ConnectDeviceAccountStepContainerProps) {
     const [title, setTitle] = useState<string>();
     const [text, setText] = useState<string>();
     const [styles, setStyles] = useState<any>({});
-    const [deviceType, setDeviceType] = useState<string>("");
-    const [providerID, setProviderID] = useState<number>();
-    const [connected, setConnected] = useState<boolean>();
+    const [deviceType, setDeviceType] = useState<string>(
+        props.deviceType || ''
+    );
+    const [loading, setLoading] = useState<boolean>(true);
+
+    function onConnect() {
+        const providerID = convertToProviderID(deviceType);
+        MyDataHelps.connectExternalAccount(providerID, {
+            openNewWindow: true,
+        }).then(function () {
+            MyDataHelps.completeStep(providerID);
+        });
+    }
+
+    function isTest() {
+        return (
+            !MyDataHelps.baseUrl ||
+            MyDataHelps.baseUrl.startsWith("https://mdhorg.ce.dev") ||
+            MyDataHelps.baseUrl.startsWith("https://mydatahelps.dev")
+        );
+    }
+
+    function convertToProviderID(deviceType: string) {
```

Comment:
Will probably refactor this to use the new methods in https://github.com/CareEvolution/MyDataHelpsUI/pull/118 so this can be centralized

Reply:
Fixed.